CardsCashCasino/Data/HoldEmUtil.cs

```diff
@@ -0,0 +1,335 @@
+        /// create and return the best possible hand and ranking.
+        /// 
+        /// In the event of a tie in rankings, the tie would be broken by calling KickerValue on each of the hands. 
+        /// If the two return values are the same, split the pot.
+        /// </summary>
+        /// <param name="community">The 5-card list of community cards</param>
+        /// <param name="cards">The 2-card list representing player's hole cards.</param>
+        /// <returns>A 5-card list representing the optimal hand, and the ranking it would give.</returns>
+        public static Tuple<List<Card>, Ranking> GetScore(List<Card> community, List<Card> cards)
+        {
+            Ranking best = Ranking.HIGH_CARD;
+            List<List<Card>> bestHands = new();
+
+            foreach (IEnumerable<Card> iCards in Get3CardPermutations(community))
+            {
+                List<Card> hand = SortedHand(iCards.Concat(cards).ToList());
+                Ranking current = GetRanking(hand);
+                if (current < best)
+                {
+                    best = current;
+                    bestHands.Clear();
+                    bestHands.Add(hand);
+                }
+                else if (current == best)
+                {
+                    bestHands.Add(hand);
+                }
+            }
+
+            long bestHandVal = 0;
+            int bestI = -1;
+
+            for (int i = 0; i < bestHands.Count; i++)
+            {
+                long handVal = KickerValue(bestHands[i]);
+                if (handVal > bestHandVal)
+                {
+                    bestHandVal = handVal;
+                    bestI = i;
+                }
+            }
+
+            return new Tuple<List<Card>, Ranking>(bestHands[bestI], best);
+        }
+
+        /// <summary>
+        /// Used for breaking a tie between two hands of equal rank.
+        /// Assigns a value to hand1 that will be greater than some hand2, if hand1 has some card C such that 
+        ///     For each card in hand2 with a greater value, hand1 has another card with equal value.  
+        /// For example, the hand [A,2,2,2,2] Is worth a greater value than [K,K,K,K,Q]
+        /// And the hand [A,A,A,K,5] is worth a greater value than [A,A,K,K,K]
+        /// Hands [5,5,4,3,2] and [5,5,4,3,2] are considered tied and should result in a split pot.
+        /// </summary>
+        /// <param name="hand">A 5-card hand that was returned by GetScore.</param>
+        /// <returns>the value that will be compared to other values returned by this function to determine a winner.</returns>
+        public static long KickerValue(List<Card> hand)
+        {
+            Debug.Assert(hand.Count == 5);
+
+            // Special case: 2345A
+            if (hand[0].GetPokerValue() == 2 && hand[1].GetPokerValue() == 3 && hand[2].GetPokerValue() == 4 && hand[3].GetPokerValue() == 5 && hand[4].IsAce())
```

Comment:
instead of using GetPokerValue() or IsAce(), just check the value enum. This might require turning it into a public property, like suit, which is totally fine.